wiki/start/a-simple-game.md

```diff
@@ -5,605 +5,1091 @@ redirect_from:
+```java
+//This is a comment. It will be ignored by the compiler.
+```
 
-If we just run the project, we will get an error: `Couldn't load file: badlogic.jpg`. Your Run Configuration has to be properly configured first: Select as working directory `PATH_TO_YOUR_PROJECT/​drop/assets`. When we run the game now, we will get the default 'game' generated by the setup app: a Badlogic Games image on a red background. Not too exciting, but that’s about to change.
+However, it is good practice to make your own comments to explain confusing code or to label parts of your design.
 
-## The Game
-The game idea is very simple:
+```java
+//Restrict bucket movement to the width of the viewport.
+```
 
-  * Catch raindrops with a bucket.
-  * The bucket is located in the lower part of the screen.
-  * Raindrops spawn randomly at the top of the screen every second and accelerate downwards.
-  * Player can drag the bucket horizontally via the mouse/touch or move it via the left and right cursor keys.
-  * The game has no end - think of it as a zen-like experience :)
+Import statements are an important part of Java programming. Thankfully, they are automatically added by modern IDE's as you type your code. They are omitted in the examples, but assume they are necessary in your code. They appear at the top of the file under the package statement.
 
-## The Assets
-We need a few images and sound effects to make the game look somewhat pretty. For the graphics we need to define a target resolution of 800x480 pixels (landscape mode on Android). If the device the game is run on does not have that resolution, we simply scale everything to fit on the screen.
+```java
+import com.badlogic.gdx.ApplicationListener;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Input.Keys;
+import com.badlogic.gdx.InputProcessor;
+import com.badlogic.gdx.audio.Music;
+import com.badlogic.gdx.audio.Sound;
+```
 
-**Note:** for high profile games you might want to consider using different assets for different screen densities. This is a big topic on its own and won't be covered here.
-{: .notice--primary}
+There are times where something named in libGDX is named the same as something found in another package. As you type `Rectangle` in Idea, for example, it will prompt you to press alt+enter to autocomplete the import. Always opt for the libGDX named class in the subsequent menu.
 
-The raindrop and the bucket should take up a small(ish) portion of the screen vertically, so we'll let them have a size of 64x64 pixels.
+![libGDX Rectangle class](/assets/images/dev/a-simple-game/1.png)
 
-The following sources provide some sample assets:
+Decimal numbers in OpenGL based games (like those made with libGDX) are usually described with floating point variables like `22.5f`. Float is preferred over Double (such as `22.5`) because it uses less memory, it's what's supported by most hardware, and it's what OpenGL usually expects.
```

Comment:
Might be worth it to mention that forgetting the `f` can cause build errors, so if you encounter those errors, check to make sure what type a function wants to receive. If it wants a float, change `1.2` to `1.2f`, and if it wants an int, `(int)myNumber` might be needed.

Reply:
I put this in my own words. Thanks.